### PR TITLE
Fct 1311 - icons bundle 2

### PR DESCRIPTION
**Pull request by ByronDWall**
Updates the SVGR template for generating icons, results in a substantial reduction in bundle size for ui kit icons, as shown in screenshots.

Implements a unique entrypoint for each icon, which enables better tree shaking in consuming applications. This results in a substantial decrease in bundle size for consuming applications, since they can bundle only necessary icons, as shown in the third screenshot.

From 1.35MB to 287kb unzipped, from 341kb to 201kb parsed, from 46kb to 38kb gzipped.

Previous:

Screenshot 2024-12-10 at 1 45 08 PM

Without unique entrypoints:
Screenshot 2024-12-10 at 1 45 05 PM

WITH UNIQUE ENTRYPOINTS FOR EACH ICON:
Screenshot 2024-12-13 at 2 31 30 PM

### CODE REVIEW

**Comment by jaikamat** [comment]
image
Amazing work 😎

**Review by jaikamat — APPROVED** [review]

**Review by tylermorrisford — APPROVED** [review]

**Review by CarlosCortizasCT — APPROVED** [review]
Awesome! 🎉

File: .github/workflows/preview-release.yml [hunk 1]
```diff
@@ -30,13 +30,13 @@ jobs:
...
           token: ${{ steps.generate_github_token.outputs.token }}
 
-      - name: Validate branch name
-        run: |
-          BRANCH_NAME=$(echo "${GITHUB_REF#refs/heads/}" | tr '/' '-')
-          if [[ ! "$BRANCH_NAME" =~ ^preview/[a-zA-Z0-9_-]+$ ]]; then
-            echo "Ignoring PR because of the branch name. Exiting workflow."
-            exit 1
-          fi
+      # - name: Validate branch name
```
Comment (tdeekens) [inline .github/workflows/preview-release.yml:33]: Should this be re-added?

File: packages/components/icons/generated/ImportReact/package.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "main": "dist/commercetools-uikit-icons-generated-ImportReact.cjs.js",
```
Comment (tdeekens) [inline packages/components/icons/generated/ImportReact/package.json:2]: Should we kebab case the package name part which is the icon?

**Review by tdeekens — COMMENTED** [review]
Thanks for going deeper on this! I cant see a diff in SVGR template strangely. By each preconstruct entry we enable the tree shake-ability?

**Comment by tdeekens** [comment]
Sorry for re-reviewing. Disregard. Just things that appear to the eye when reviewing thrice ;)

File: .github/workflows/preview-release.yml (hunk 1 above)
Comment (ByronDWall) [reply .github/workflows/preview-release.yml:33]: yes, although it should be updated/fixed in another PR

File: packages/components/icons/package.json [hunk 3]
```diff
@@ -22,10 +22,17 @@
       "./index.ts",
       "./custom-icon/index.ts",
       "./inline-svg/index.ts",
-      "./leading-icon/index.ts"
+      "./leading-icon/index.ts",
+      "./generated/*.tsx"
```
Comment (ByronDWall) [inline packages/components/icons/package.json:26]: This glob tells preconstruct that each of the icons should have an entrypoint in the package.

File: packages/components/icons/package.json [hunk 4]
```diff
@@ -22,10 +22,17 @@
...
     ]
   },
-  "files": ["dist", "custom-icon", "inline-svg", "leading-icon"],
+  "files": [
+    "dist",
+    "custom-icon",
+    "inline-svg",
+    "leading-icon",
+    "generated/**"
```
Comment (ByronDWall) [inline packages/components/icons/package.json:34]: This glob tells preconstruct that we expect the dist/package.json for each icon to be in the `generated` folder.

File: packages/components/icons/generated/ImportReact/package.json (hunk 2 above)
Comment (ByronDWall) [reply packages/components/icons/generated/ImportReact/package.json:2]: everything in `packages/icons/generated` is auto-generated by preconstruct when you run `preconstruct dev` or `preconstruct build`, and there is no way to configure this.

**Pull request merged**